src/storage/errors.rs

```diff
@@ -197,6 +208,11 @@ pub fn extract_region_error<T>(res: &Result<T>) -> Option<errorpb::Error> {
             err.set_message("TiKV is Closing".to_string());
             Some(err)
         }
+        Err(Error(box ErrorInner::DeadlineExceeded)) => {
```

Comment:
It's possible the client receives the error and it's a region error. TiDB will invalidate the cache and backoff. I think we should distinguish it.

Reply:
Never mind. TiDB can compare the error message for now.

Reply:
I think this PR does not affect current TiDB now because the field in `Context` is not set.

I want to change the TiDB handling of this later. The simplest handling is make client timeout a bit shorter than TiKV's timeout, then the timeout error will never be received by TiDB. The timeout in TiKV just prevents waste of resource.

Reply:
Agree with it.